bgpd: add display support for EVPN extended community subtypes

jaredmauch opened the pull request:
Add display formatting for several EVPN extended community subtypes that were previously not displayed in a human-readable format:

- E-Tree Extended Community (RFC 8317, subtype 0x05)
- I-SID Extended Community (draft-ietf-bess-evpn-virtual-eth-segment, subtype 0x07)
- Load Balancing Extended Community (RFC 9014, subtype 0x0e)

The Load Balancing Extended Community is standardized in RFC 9014 as EVPN encoding type, but earlier drafts used OPAQUE encoding. The code handles both encoding types for backward compatibility.

Also reorder DEF_GW subtype definition to maintain proper ordering and move its display handling from OPAQUE to EVPN section where it belongs.

ton31337 reviewed (commented):
Can we test it?

ton31337 commented on tests/topotests/lib/bgp.py:
```diff
@@ -1334,7 +1334,7 @@ def verify_router_id(tgen, topo, input_dict, expected=True):
 
 @retry(retry_timeout=150)
-def verify_bgp_convergence(tgen, topo=None, dut=None, expected=True, addr_type=None):
+def verify_bgp_convergence(tgen, topo=None, dut=None, expected=True, addr_type=None, seconds_left=None):
```
This should be on a separate commit.

ton31337 commented on tests/topotests/bgp_evpn_extcommunity_subtypes/test_bgp_evpn_extcommunity_subtypes.py:
```diff
@@ -0,0 +1,386 @@
+
+    for rname, router in router_list.items():
+        router.load_config(
+            TopoRouter.RD_ZEBRA, os.path.join(CWD, "{}/zebra.conf".format(rname))
```
Let's use frr.conf instead of per-daemon configurations.

ton31337 reviewed (commented):
Also, drop the merge commit, please.

ton31337 commented:
Drop the merge commit and squash the last commit to the first one.

jaredmauch committed:
bgpd: add display support for EVPN extended community subtypes

Add display formatting for several EVPN extended community subtypes
that were previously not displayed in a human-readable format:

- E-Tree Extended Community (RFC 8317, subtype 0x05)
- I-SID Extended Community (draft-ietf-bess-evpn-virtual-eth-segment,
  subtype 0x07)
- Load Balancing Extended Community (RFC 9014, subtype 0x0e)

The Load Balancing Extended Community is standardized in RFC 9014 as
EVPN encoding type, but earlier drafts used OPAQUE encoding. The code
handles both encoding types for backward compatibility.

Also reorder DEF_GW subtype definition to maintain proper ordering
and move its display handling from OPAQUE to EVPN section where it
belongs.

Add comprehensive topotests to verify parsing and display of these
EVPN extended community subtypes across multiple route types:

- IPv4 unicast routes
- IPv6 unicast routes
- IPv4 EVPN Type-2 routes (MAC/IP)
- IPv6 EVPN Type-2 routes (MAC/IPv6)

The tests use ExaBGP to inject routes with the various EVPN extended
community subtypes and verify they are correctly parsed and displayed
by FRR. The test infrastructure is also enhanced to support verification
of extendedCommunity attributes in the verify_bgp_community function.

Signed-off-by: jared mauch <jared@puck.nether.net>
